WcaOnRails/app/models/competition.rb

```diff
@@ -305,6 +305,18 @@ def events
     eventSpecs.split.map { |e| Event.find_by_id(e.split("=")[0]) }.sort_by &:rank
   end
 
+  def has_event?(event_id)
+    event_id.in?(self.events.map{ |ev| ev.id })
+  end
+
+  def belongs_to_region?(region_or_country)
+    (self.countryId == region_or_country) || (Country.find(self.countryId).continentId == region_or_country)
```

Comment:
Added this and removed the def country_name, and change the place where it was being called.